pallets/nominee-election/src/lib.rs

```diff
@@ -0,0 +1,280 @@
+use frame_system::pallet_prelude::*;
+pub use pallet::*;
+
+use sp_std::{convert::TryInto, vec::Vec};
+
+/// Nominee Election Coefficients
+/// https://docs.parallel.fi/dev/staking/staking-election
+#[derive(Encode, Decode, Eq, PartialEq, Clone, RuntimeDebug, Default)]
+#[cfg_attr(feature = "std", derive(serde::Deserialize, serde::Serialize))]
+pub struct NomineeScoreCoefficients {
+    // R: Reputation, 0 or 1
+    // CR: Commission Rate
+    // N: Nomination of one validator
+    // EEP: Average Era Points of one validator in the past week.
+    // EEPA: Average Era Points of All validators in the past week.
+    // SR: Slash Record, default 1, set to 0 if ever slashed in the past month.
+
+    // Commission rate factor
+    // A constant shows how much influence of the Commission Rate to validator's final score
+    pub crf: u32,
+    // Nomination factor
+    // A constant shows how much influence of the Nomination to validator's final score
+    pub nf: u32,
+    // Era Points factor
+    // A constant shows how much influence of the Era Points to validator's final score
+    pub epf: u32,
+    //
+    // Score: R * (CRF * (1 - CR) + NF * (1 / N) + EPF * (EEP / EEPA)) * SR
+}
+
+/// Info of the validator to be elected
+#[derive(Encode, Decode, Eq, PartialEq, Clone, RuntimeDebug, Default)]
+#[cfg_attr(feature = "std", derive(serde::Deserialize, serde::Serialize))]
+pub struct ValidatorInfo<AccountId> {
+    // Account Id
+    pub account_id: AccountId,
+    // Nomination (token amount)
+    pub stakes: u128,
+    // Score
+    pub score: u128,
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+
+    use super::*;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        /// The origin which can update staking election coefficients
+        type UpdateOrigin: EnsureOrigin<Self::Origin>;
+
+        /// The origin which can feed whitelisted validators
+        type WhitelistUpdateOrigin: EnsureOrigin<Self::Origin>;
+
+        /// The maximum size of selected validators
+        #[pallet::constant]
+        type MaxNumValidators: Get<u32>;
+
+        /// Approved accouts which can set validators
+        type Members: SortedMembers<Self::AccountId>;
+    }
+
+    /// Nominee election coefficients
+    #[pallet::storage]
+    #[pallet::getter(fn coefficients)]
+    pub type Coefficients<T: Config> = StorageValue<_, NomineeScoreCoefficients, ValueQuery>;
+
+    /// Validators selected by off-chain client
+    #[pallet::storage]
+    #[pallet::getter(fn validators)]
+    pub type Validators<T: Config> =
+        StorageValue<_, BoundedVec<ValidatorInfo<T::AccountId>, T::MaxNumValidators>, ValueQuery>;
+
+    /// Whitelisted validators selected by council
+    #[pallet::storage]
+    #[pallet::getter(fn whitelisted_validators)]
+    pub type WhitelistedValidators<T: Config> =
```

Comment:
WhitelistedValidators can be combined with `Validators` with a flag.
actually there is no need to add WhitelistedValidators now.

Reply:
the problem is that whitelisted validators normally won't have a score etc because it's feeded from off-chain